### PR TITLE
clang-tidy: use nullptr where appropriate

### PR DESCRIPTION
**Pull request by neheb**
Found with modernize-use-nullptr

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: include/exiv2/types.hpp [hunk 1]
```diff
@@ -242,7 +242,7 @@ namespace Exiv2 {
         void free();
 
         //! Reset value
-        void reset(std::pair<byte*, size_t> =std::make_pair((byte*)(0),size_t(0)));
+        void reset(std::pair<byte*, size_t> =std::make_pair((byte*)nullptr,size_t(0)));
```
Comment (piponazo) [inline include/exiv2/types.hpp:245]: It should be possible to remove from here the cast `(byte *)` from inside the make_pair call, no ?

**Review by piponazo — APPROVED** [review]
Thanks for the contribution!

It definitely helps to improve the code readability in some places.

File: include/exiv2/types.hpp (hunk 1 above)
Comment (neheb) [reply include/exiv2/types.hpp:245]: Beats me. This was an automated change. I am also unfamiliar with this codebase.

Comment (piponazo) [reply include/exiv2/types.hpp:245]: No problem, I realised about that later. Let's merge the PR as it is, since it is a useful contribution 😉

**Pull request merged**